os/binfmt/binfmt_execmodule.c

```diff
@@ -325,33 +264,36 @@ int exec_module(FAR struct binary_s *binp)
 	 */
 
 	if (binp->nctors > 0) {
-		task_starthook(tcb, exec_ctors, (FAR void *)binp);
+		task_starthook(newtcb, exec_ctors, (FAR void *)binp);
 	}
 #endif
 
 	/* Get the assigned pid before we start the task */
 
-	pid = tcb->cmn.pid;
+	pid = newtcb->cmn.pid;
 
-#ifdef CONFIG_APP_BINARY_SEPARATION
 	rtcb->uheap = 0;
+#ifdef CONFIG_ARM_MPU
+	/* Reset the mpu_regs in parent task to zero */
+	memset(rtcb->mpu_regs, 0, sizeof(rtcb->mpu_regs));
+#endif
```

Comment:
Could you review to set new_tcb here instead of getting them from rtcb and clearing?

Reply:
If you want to keep original, then let's leave why parent kernel thread needs to be initialized.